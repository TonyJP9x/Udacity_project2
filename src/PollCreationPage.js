import React, { useEffect, useState } from 'react';
import NavBar from './NavBar';
import { useDispatch, useSelector } from 'react-redux';
import { saveQuestion } from './Slices/QuestionSlice';
import { useNavigate } from 'react-router-dom';

function PollCreationPage(props) {
    const navigate = useNavigate()
    const dispatch = useDispatch()
    const userInfo = useSelector((state) => state.login.value);
    useEffect(() => {
        if(!userInfo.id){
            navigate('/')
        }
    },[])
    const author = userInfo.id
    const [optionOne, setOptionOne] = useState('')
    const [optionTwo, setOptionTwo] = useState('')
    const handleOnSubmit = (e) =>{
        e.preventDefault()
    dispatch(saveQuestion({optionOneText:optionOne, optionTwoText:optionTwo, author}))
    if(optionOne !== null && optionTwo !== null){

        setTimeout(() => {
            navigate('/Home')
        },1000)
    }
    }


    return (
        <div>
            <NavBar />
             <form >
                <h1>Would you rather</h1>
                <h3>Create your own poll</h3>
                <h4>First Option</h4>
                <input data-testid='first-option' type='text' required placeholder='Option One' style={{width:'500px'}} value={optionOne} onChange={(e) => setOptionOne(e.target.value)}/>
                <h4>Second Option</h4>
                <input data-testid='second-option' type='text' required placeholder='Option Two' style={{width:'500px'}} value={optionTwo} onChange={(e) => setOptionTwo(e.target.value)} /><br/>
              <button data-testid='submit-button' type="submit" className="btn btn-primary" onClick={(e) => handleOnSubmit(e)}  style={{marginTop: '20px', marginBottom: '20px'}}>Submit</button>
                
           
    
             
            </form>
        </div>
    );
}

export default PollCreationPage;