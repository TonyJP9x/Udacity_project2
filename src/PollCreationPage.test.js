import PollCreationPage from "./PollCreationPage";
import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import store from "./Store";
import * as React from "react";
import { waitFor } from "@testing-library/react";


describe("PollCreationPage", () => {
    it('will have all expected field', () => {
        var component = render(
            <MemoryRouter>
              <Provider store={store}>
                <PollCreationPage />
              </Provider>
            </MemoryRouter>
          );
          expect(component).toMatchSnapshot();
    })
  
});
