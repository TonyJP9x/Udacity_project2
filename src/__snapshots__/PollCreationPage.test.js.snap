// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollCreationPage will have all expected field 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="margin-bottom: 2px;"
        >
          <nav
            class="navbar navbar-expand-lg navbar-light bg-light"
          >
            <div
              class="row"
              style="width: 100%;"
            >
              <div
                class="col-10"
                style="display: flex; padding-left: 30px;"
              >
                <a
                  class="navbar-brand"
                  href="/home"
                >
                  Home
                </a>
                <a
                  class="navbar-brand"
                  href="/leaderboard"
                >
                  LeaderBoard
                </a>
                <a
                  class="navbar-brand"
                  href="/add"
                >
                  New
                </a>
              </div>
              <div
                class="col-2"
                style="display: flex;"
              >
                <a
                  class="navbar-brand"
                  href="/"
                >
                   
                </a>
                <img
                  style="width: 40px; border-radius: 50%; margin-right: 20px;"
                />
                <a
                  class="navbar-brand"
                  href="/"
                >
                  Logout
                </a>
              </div>
            </div>
          </nav>
        </div>
        <form>
          <h1>
            Would you rather
          </h1>
          <h3>
            Create your own poll
          </h3>
          <h4>
            First Option
          </h4>
          <input
            data-testid="first-option"
            placeholder="Option One"
            required=""
            style="width: 500px;"
            type="text"
            value=""
          />
          <h4>
            Second Option
          </h4>
          <input
            data-testid="second-option"
            placeholder="Option Two"
            required=""
            style="width: 500px;"
            type="text"
            value=""
          />
          <br />
          <button
            class="btn btn-primary"
            data-testid="submit-button"
            style="margin-top: 20px; margin-bottom: 20px;"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="margin-bottom: 2px;"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="row"
            style="width: 100%;"
          >
            <div
              class="col-10"
              style="display: flex; padding-left: 30px;"
            >
              <a
                class="navbar-brand"
                href="/home"
              >
                Home
              </a>
              <a
                class="navbar-brand"
                href="/leaderboard"
              >
                LeaderBoard
              </a>
              <a
                class="navbar-brand"
                href="/add"
              >
                New
              </a>
            </div>
            <div
              class="col-2"
              style="display: flex;"
            >
              <a
                class="navbar-brand"
                href="/"
              >
                 
              </a>
              <img
                style="width: 40px; border-radius: 50%; margin-right: 20px;"
              />
              <a
                class="navbar-brand"
                href="/"
              >
                Logout
              </a>
            </div>
          </div>
        </nav>
      </div>
      <form>
        <h1>
          Would you rather
        </h1>
        <h3>
          Create your own poll
        </h3>
        <h4>
          First Option
        </h4>
        <input
          data-testid="first-option"
          placeholder="Option One"
          required=""
          style="width: 500px;"
          type="text"
          value=""
        />
        <h4>
          Second Option
        </h4>
        <input
          data-testid="second-option"
          placeholder="Option Two"
          required=""
          style="width: 500px;"
          type="text"
          value=""
        />
        <br />
        <button
          class="btn btn-primary"
          data-testid="submit-button"
          style="margin-top: 20px; margin-bottom: 20px;"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
